### PR TITLE
Fix role facts when using multiple plays

### PR DESCRIPTION
**Pull request by hackallcode**
Before this patch, variables were stored in the first play and continued to be used in the second play.
So, it was impossible to redefine variable by `vars` section.

Now the variables are saved to the dictionary, so they do not affect the next play.

### CODE REVIEW

File: molecule/update_cartridge/molecule.yml [hunk 1]
```diff
@@ -49,6 +49,7 @@ scenario:
     - syntax
     - create
     - converge
+    # - idempotence
```
Comment (dokshina) [inline molecule/update_cartridge/molecule.yml:52]: Debug?

File: molecule/check_facts/tasks/check.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+
+- fail:
+    msg: "Task 'check_task' wasn't started"
```
Comment (dokshina) [inline molecule/check_facts/tasks/check.yml]: Prepare wasn't started?

File: molecule/check_facts/tasks/check.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  when: delivered_package_path_from_check_task is not defined
+  run_once: true
+
+- fail:
+    msg: "Incorrect value of 'delivered_package_path', got: '{{ delivered_package_path_from_check_task }}'"
+  when: delivered_package_path_from_check_task != "my_task_default"
```
Comment (dokshina) [inline molecule/check_facts/tasks/check.yml]: `delivered_package_path != "my_task_default"`

Comment (dokshina) [reply molecule/check_facts/tasks/check.yml]: Oh no. I can't understand what does this test do...

File: molecule/check_facts/tasks/prepare_to_check.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+
+- set_fact:
```
Comment (dokshina) [inline molecule/check_facts/tasks/prepare_to_check.yml]: `prepare_to_check.yml` name confuses - I was thinking about molecule prepare.
What about `set_delivered_package_path_step`?

File: molecule/update_cartridge/molecule.yml (hunk 1 above)
Comment (hackallcode) [reply molecule/update_cartridge/molecule.yml:52]: No, it's to show that this step is skipped here

File: molecule/check_facts/tasks/check.yml (hunk 2 above)
Comment (hackallcode) [reply molecule/check_facts/tasks/check.yml]: This task wasn't started, prepare will be started in any case)

File: molecule/check_facts/tasks/check.yml (hunk 3 above)
Comment (hackallcode) [reply molecule/check_facts/tasks/check.yml]: It's test that variable delivered_package_path is empty

File: molecule/check_facts/tasks/prepare_to_check.yml (hunk 4 above)
Comment (hackallcode) [reply molecule/check_facts/tasks/prepare_to_check.yml]: But it's possible to make some other tasks here in future

File: molecule/check_facts/converge.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  hosts: cluster
+  gather_facts: false
+  tasks:
+    - import_role:
+        name: ansible-cartridge
+      vars:
+        cartridge_scenario_name: 'custom_scenario'
+
+    - fail:
+        msg: "Scenario variable wasn't reset by last role import"
```
Comment (dokshina) [inline molecule/check_facts/converge.yml]: `cartridge_scenario variable wasn't reset by last role import`.

File: molecule/check_facts/hosts.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    # common cartridge opts
+    cartridge_app_name: myapp
+    cartridge_cluster_cookie: secret-cookie
+
+    cartridge_custom_steps_dir: './tasks'
+    cartridge_custom_scenarios:
+      custom_scenario:
+        - custom_task
+
+  hosts:
```
Comment (dokshina) [inline molecule/check_facts/hosts.yml:18]: Seems that all this hosts are redundant except one. The same for replicaset group.

File: molecule/check_facts/hosts.yml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    ansible_user: root
+    ansible_connection: docker
+    become: true
+    become_user: root
+
+    # common cartridge opts
+    cartridge_app_name: myapp
+    cartridge_cluster_cookie: secret-cookie
+
+    cartridge_custom_steps_dir: './tasks'
```
Comment (dokshina) [inline molecule/check_facts/hosts.yml]: Why don't you call steps dir `steps`?

**Review by dokshina — DISMISSED** [review]
Please, update Changelog before merge.

**Review by dokshina — DISMISSED** [review]

**Review by dokshina — DISMISSED** [review]

**Review by dokshina — APPROVED** [review]

**Pull request merged**